diff --git a/Source/ZipLib/detail/ZipCentralDirectoryFileHeader.cpp b/Source/ZipLib/detail/ZipCentralDirectoryFileHeader.cpp
index a204ed0..a567a0d 100644
--- a/Source/ZipLib/detail/ZipCentralDirectoryFileHeader.cpp
+++ b/Source/ZipLib/detail/ZipCentralDirectoryFileHeader.cpp
@@ -55,7 +55,7 @@ bool ZipCentralDirectoryFileHeader::Deserialize(std::istream& stream)
   if (stream.fail() || Signature != SignatureConstant)
   {
     stream.clear();
-    stream.seekg(static_cast<std::ios::off_type>(static_cast<std::ios::streamoff>(stream.tellg()) - stream.gcount()), std::istream::beg);
+    stream.seekg(static_cast<std::ios::off_type>(static_cast<std::streamoff>(stream.tellg()) - stream.gcount()), std::istream::beg);
     return false;
   }
 
diff --git a/Source/ZipLib/detail/ZipLocalFileHeader.cpp b/Source/ZipLib/detail/ZipLocalFileHeader.cpp
index 4fdbac8..7976480 100644
--- a/Source/ZipLib/detail/ZipLocalFileHeader.cpp
+++ b/Source/ZipLib/detail/ZipLocalFileHeader.cpp
@@ -53,7 +53,7 @@ bool ZipLocalFileHeader::Deserialize(std::istream& stream)
   if (stream.fail() || Signature != SignatureConstant)
   {
     stream.clear();
-    stream.seekg(static_cast<std::ios::off_type>(static_cast<std::ios::streamoff>(stream.tellg()) - stream.gcount()), std::ios::beg);
+    stream.seekg(static_cast<std::ios::off_type>(static_cast<std::streamoff>(stream.tellg()) - stream.gcount()), std::ios::beg);
     return false;
   }
 
